backbone.js

```diff
@@ -1427,6 +1428,9 @@
   var splatParam    = /\*\w+/g;
   var escapeRegExp  = /[\-{}\[\]+?.,\\\^$|#\s]/g;
 
+  // Route handlers registered through Router instances are stored here.
+  Router.handlers = [];
```

Comment:
I'm not a fan of this. This just moves the singleton responsibility from `history` to here.